pkg/live/load.go

```diff
@@ -154,16 +113,25 @@ func readInvInfoFromStream(in io.Reader) (kptfilev1.Inventory, error) {
 		return kptfilev1.Inventory{}, err
 	}
 
-	if len(invFilter.Inventories) > 1 ||
-		len(rgFilter.Inventories) > 1 ||
-		(len(invFilter.Inventories) > 0 && len(rgFilter.Inventories) > 0) {
-		return kptfilev1.Inventory{}, &MultipleInventoryInfoError{}
+	// Ensure only exactly 1 inventory exists and surface the correct type of error.
+	// Multiple Kptfile inventories found.
+	if len(invFilter.Inventories) > 1 {
+		return kptfilev1.Inventory{}, &pkg.MultipleKfInv{}
+	}
+	// Multiple ResourceGroup inventories found.
+	if len(rgFilter.Inventories) > 1 {
+		return kptfilev1.Inventory{}, &pkg.MultipleResourceGroupsError{}
+	}
+	// Multiple inventories found in Kptfile and ResourceGroup objects.
+	if len(invFilter.Inventories) > 0 && len(rgFilter.Inventories) > 0 {
+		return kptfilev1.Inventory{}, &pkg.MultipleInventoryInfoError{}
 	}
```

Comment:
I'm thinking that after separating inventory from kptfile, the kptfile is no longer required by `kpt live apply` so the content of Kptfile is no longer needed to be verified. 

Also, verifying the Kptfile and RG is complicated: e.g. an inventory error requires migration plans, if more than one inventory or/and RG are detected, it's hard for both `kpt live migrate` and users to make good decisions to recover. So I think an easier approach would be: 
- `kpt live apply` ignores what Kptfile says (this also resolves the problems that nested Kptfile files may have different inventory)
- resourceGroup is a meta resource, but it is passed into `kpt live` as a normal KRM resource.  One and only one RG is required in `kpt live apply`

Reply:
So even with this change we want to let users continue to use inventory from the Kptfile if they want to. We don't want to make this a change that requires users to do the migration in order to do a successful apply. For this particular check, I think we should fail if we find inventory information in both the Kptfile and the resourcegroup file. There isn't any good way to know which one if the correct one, so we need to user to clean it up.

Reply:
The feedback I got from skaffold (when introducing the inventory and have inventory-id conflict) is that cleaning up the duplicate inventory is challenging for users because 1. kpt inits the inventory by default, so users may lack knowledge about how RG is created and what it is used for. 2. users do not know what's the current in-use RG is and it is hard to find it out from the cluster. 3. If user force a new RG, they do not know how to generate a new one (is it random string or has to much some specific format). 
I'd propose make users life easier by giving some direct suggestions like "delete the inventory in Kptfile" or "use --force to override" etc.

Reply:
Yeah, agree the error messages could be better. We can provide custom error handlers to provide more information. I'll handle that in a follow-up since this is already a pretty massive PR.